Explorer/Assets/Scripts/ECS/Unity/PrimitiveColliders/Systems/InstantiatePrimitiveColliderSystem.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CrdtEcsBridge.Components.Defaults;
+using CrdtEcsBridge.Components.Transform;
+using CrdtEcsBridge.Physics;
+using DCL.ECSComponents;
+using ECS.Abstract;
+using ECS.ComponentsPooling;
+using ECS.Unity.Groups;
+using ECS.Unity.PrimitiveColliders.Components;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ECS.Unity.PrimitiveColliders.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    public partial class InstantiatePrimitiveColliderSystem : BaseUnityLoopSystem
+    {
+        private const int LAYER_PHYSICS = (int)ColliderLayer.ClPhysics;
+        private const int LAYER_POINTER = (int)ColliderLayer.ClPointer;
+        private const int LAYER_PHYSICS_POINTER = LAYER_PHYSICS | LAYER_POINTER;
+
+        private readonly IComponentPoolsRegistry poolsRegistry;
+
+        private static readonly Dictionary<PBMeshCollider.MeshOneofCase, ISetupCollider> SETUP_COLLIDER_LOGIC = new ()
+        {
+            { PBMeshCollider.MeshOneofCase.Box, new SetupBoxCollider() },
+            { PBMeshCollider.MeshOneofCase.Sphere, new SetupSphereCollider() },
+            { PBMeshCollider.MeshOneofCase.Cylinder, new SetupCylinderCollider() },
+            { PBMeshCollider.MeshOneofCase.Plane, new SetupPlaneCollider() },
+        };
+
+        private readonly Dictionary<PBMeshCollider.MeshOneofCase, ISetupCollider> setupColliderCases;
+
+        internal InstantiatePrimitiveColliderSystem(World world, IComponentPoolsRegistry poolsRegistry,
+            Dictionary<PBMeshCollider.MeshOneofCase, ISetupCollider> setupColliderCases = null) : base(world)
+        {
+            this.setupColliderCases = setupColliderCases ?? SETUP_COLLIDER_LOGIC;
+            this.poolsRegistry = poolsRegistry;
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateNonExistingColliderQuery(World);
+            TrySetupExistingColliderQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBMeshCollider), typeof(SDKTransform), typeof(Transform))]
+        [None(typeof(PrimitiveColliderComponent))]
+        private void InstantiateNonExistingCollider(in Entity entity, ref PBMeshCollider sdkComponent, ref Transform transform)
+        {
+            var component = new PrimitiveColliderComponent();
```

Comment:
How would the release flow work for this component?

When the world gets destroyed, they would just get deleted alongside their referenced unity colliders? Should we consider sending thos unity colliders back to the pool before they get destroyed?

Reply:
I missed that, I introduced a concept of `PoolableProviders` along the generic system that handles them

Reply:
Take a look at ` ReleasePoolableComponentSystem<TProvider>`, I believe it will be useful for further implementations as well